app/services/service.ts

```diff
@@ -81,6 +85,11 @@ export abstract class Service {
     if (enforcer !== singletonEnforcer) throw 'Cannot construct singleton';
   }
 
+  /**
+   * calls only once per application life
+   */
+  protected beforeInit() {}
```

Comment:
couldn't any of the operations that would need to be here just be included in `init` at the top of that function?

Reply:
Looks like its something that I forgot to delete after my experiments with `Vuex` mutations inheritance. Will remove